test/envtest/konnect_entities_kongvault_test.go

```diff
@@ -121,4 +123,57 @@ func TestKongVault(t *testing.T) {
 			assert.True(c, factory.SDK.VaultSDK.AssertExpectations(t))
 		}, waitTime, tickTime)
 	})
+
+	t.Run("Should correctly handle conflict on create", func(t *testing.T) {
+		const (
+			vaultBackend   = "env-conflict"
+			vaultPrefix    = "env-vault-conflict"
+			vaultRawConfig = `{"prefix":"env_vault_conflict"}`
+			vaultID        = "vault-conflict-id"
+		)
+
+		t.Log("Setting up mock SDK for vault creation with conflict")
+		sdk.VaultSDK.EXPECT().CreateVault(mock.Anything, cp.GetKonnectStatus().GetKonnectID(), mock.MatchedBy(func(input sdkkonnectcomp.VaultInput) bool {
+			return input.Name == vaultBackend && input.Prefix == vaultPrefix
+		})).Return(&sdkkonnectops.CreateVaultResponse{
+			Vault: &sdkkonnectcomp.Vault{
+				ID: lo.ToPtr(vaultID),
+			},
+		}, &sdkkonnecterrs.ConflictError{})
```

Comment:
This should return just the error, no? Secondly, I believe this API won't return Conflict error but SDKError as for other entities living inside the CP.

```suggestion
		})).Return(nil, &sdkkonnecterrs.ConflictError{})
```

Reply:
bd9a356